### PR TITLE
Updated User Management and Comments Module with unit testing

### PR DESCRIPTION
**Pull request by Srishtirupa3**

### CODE REVIEW

File: UserManagement_module.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+app = Flask(__name__)
+cors = CORS(app)
+CORS(app, origins='*')
+bcrypt = Bcrypt(app)
+
+###########################################################################################################
+                        #to add new user(authority of  alpha user)
+###########################################################################################################
+
+#import re
```
Comment (Chandrashekhar-Bhavsar) [inline UserManagement_module.py:28]: remove unwanted comments

File: UserManagement_module.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+        if "email_id" not in data:
+            return jsonify({"error": "Missing 'email_id' in request data"}), 400
+        
+
+        project_id=data['project_id']
+        
+        email_id=data['email_id']
+        #user_ID =data["user_ID"]
+
```
Comment (Chandrashekhar-Bhavsar) [inline UserManagement_module.py:195]: remove unwanted lines

File: UserManagement_module.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        logging.debug(dt_string + " checking if the project manager role already existing or not since tere can be only one project manager per project....")
+        
+            
+        logging.debug(dt_string + " calling user_assign function to update the databse....")
+            
+        return user_assign(project_id,email_id)
+
+    except KeyError as e:
+        # Handle missing key in the request data
+
```
Comment (Chandrashekhar-Bhavsar) [inline UserManagement_module.py:216]: Remove unwanted lines

File: UserManagement_module.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        print("An error occurred: " + str(e))
+        
+        return jsonify({"error": "An error occurred: " + str(e)}), 500
+    
+
+############################################################################################################
+                            # API to show all users to added     
+############################################################################################################
+
+
```
Comment (Chandrashekhar-Bhavsar) [inline UserManagement_module.py:240]: Add only two blank line before function

File: app.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+@app.route('/add_user', methods=['POST'])
+def add_user():
+    return adduser()
+    
+
+@app.route('/assign_user', methods=['POST'])
+def assign_user():
+   print("hello")
+   return assignuser()   
+
```
Comment (Chandrashekhar-Bhavsar) [inline app.py:31]: Please do uniform line spacing

File: app.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import mysql.connector
+from flask_cors import CORS,cross_origin
+import bcrypt
+from flask_bcrypt import bcrypt
+from connection import *
+from queries import *
+from Comments_Module import *
+from UserManagement_module import *
+import datetime
+from datetime import datetime
```
Comment (Chandrashekhar-Bhavsar) [inline app.py:11]: use one import statement

File: db.config [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "host" : "localhost",
```
Comment (Chandrashekhar-Bhavsar) [inline db.config:2]: please do openshift pod configuration

File: myfile.txt
Comment (Chandrashekhar-Bhavsar) [inline myfile.txt:1]: remove file

File: test_for_updateprojectwisecomments.py [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+        print("#"*50)
+        print("Test case id: TCUPWC01")
+        print("Test case name: Checking for CORRECT data  ")
+        print("Inside the Test Case ")
+
+
+        # Define the API endpoint for adding a new user
+        API_ENDPOINT_post = api_url +"update_projectwise_comments"
+
+        # Prepare the data for the new user
```
Comment (Chandrashekhar-Bhavsar) [inline test_for_updateprojectwisecomments.py:23]: Uniform spacing

**Review by Chandrashekhar-Bhavsar — CHANGES_REQUESTED** [review]
Do Changes as per noted

**Review by Chandrashekhar-Bhavsar — APPROVED** [review]

**Review by Chandrashekhar-Bhavsar — CHANGES_REQUESTED** [review]
do changes

File: UserManagement_module.py (hunk 1 above)
Comment (Srishtirupa3) [reply UserManagement_module.py:28]: ok removed

File: UserManagement_module.py [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+            return jsonify({"error": "Missing 'name' in request data"}), 400
+        if "email_id" not in data:
+            return jsonify({"error": "Missing 'email_id' in request data"}), 400
+        if "contact" not in data:
+            return jsonify({"error": "Missing 'contact' in request data"}), 400
+        if "role" not in data:
+            return jsonify({"error": "Missing 'role' in request data"}), 400
+
+        name = data['name']
+
```
[the remark this answers is not in the document]
Comment (Srishtirupa3) [reply UserManagement_module.py:68]: that spacing around the operator is required as per the PEP 8 rules

File: UserManagement_module.py (hunk 2 above)
Comment (Srishtirupa3) [reply UserManagement_module.py:195]: those lines are given for proper segregation and code readability

File: UserManagement_module.py (hunk 3 above)
Comment (Srishtirupa3) [reply UserManagement_module.py:216]: Those lines are for readability

File: UserManagement_module.py (hunk 4 above)
Comment (Srishtirupa3) [reply UserManagement_module.py:240]: Noted

File: app.py (hunk 5 above)
Comment (Srishtirupa3) [reply app.py:31]: sorry, i'm not getting you. please elaborate

File: app.py (hunk 6 above)
Comment (Srishtirupa3) [reply app.py:11]: But need to import multiple things

File: db.config (hunk 7 above)
Comment (Srishtirupa3) [reply db.config:2]: ok noted

File: myfile.txt
Comment (Srishtirupa3) [reply myfile.txt:1]: removed

File: new.txt
[the remark this answers is not in the document]
Comment (Srishtirupa3) [reply new.txt:1]: removed

File: txt.txt
[the remark this answers is not in the document]
Comment (Srishtirupa3) [reply txt.txt:1]: ok...removed

**Pull request closed without merging**